app/api/league/route.ts

```diff
@@ -18,30 +24,18 @@ const LeagueConfig = z.object({
 export async function POST(request: NextRequest) {
     // check auth
     const body = await request.json();
-    if (!body.token) {
-        return NextResponse.json({"error": unauthenticatedErrorMessage}, {status: 401})
-    }
-    const client = new OAuth2Client();
-    const clientId = process.env.GOOGLE_LOGIN_CLIENT_ID;
-    let authResponse = null;
-    try {
-        authResponse = await client.verifyIdToken({
-            idToken: body.token,
-            audience: clientId
-        });
-    }
-    catch(error) {
-        console.log(error);
-        return NextResponse.json({"error": unauthenticatedErrorMessage}, {status: 401})
-    }
-
-    const payload:TokenPayload | undefined = authResponse.getPayload();
-
-    if(payload){
-        const googleUserId = payload["sub"];
-        if (googleUserId !== "108251633753098119380") {
-            return NextResponse.json({"error": "you are not authorized to perform that action"}, {status: 403});
+    const sessionCookie = request.cookies.get("session");
+    if(sessionCookie){
+        const decryptedSessionCookie = await decrypt(sessionCookie?.value) as decryptionPayload;
+        const googleUserId = decryptedSessionCookie?.sub;
+        const allowedGoogleUserIds = ["108251633753098119380", "117801378252057178101"];
+        const invalidGoogleUserId = !googleUserId || allowedGoogleUserIds.indexOf(googleUserId) < 0;
+        const invalidGoogleUserSub = Object.keys(decryptedSessionCookie).length !== 3;
+        if (invalidGoogleUserId || invalidGoogleUserSub ) {
+            return NextResponse.json({"error": "you are not authorized to perform that action"}, {status: 401});
```

Comment:
I would move a any of these lines which have to do with authorization to a separate code section so that we can keep the authentication and the authorization part's of the code separate. Removing those parts from here would look something like this:

```suggestion
        const invalidGoogleUserId = !googleUserId
        const invalidGoogleUserSub = Object.keys(decryptedSessionCookie).length !== 3;
        if (invalidGoogleUserId || invalidGoogleUserSub ) {
            return NextResponse.json({"error": unauthenticatedErrorMessage}, {status: 401});
```

And then another block which has the following some of the following code:
```
        const allowedGoogleUserIds = ["108251633753098119380", "117801378252057178101"];
        const isUserAllowed = allowedGoogleUserIds.indexOf(googleUserId) < 0;
```

This way in the latter block we can continue to return a 403, which is proper.